services/vmagent-config-update/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"time"
+
+	"prometheus-benchmark/services/vmagent-config-update/controllers"
+	"prometheus-benchmark/services/vmagent-config-update/models"
+	"prometheus-benchmark/services/vmagent-config-update/runner"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/envflag"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/logger"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/procutil"
+)
+
+var (
+	listenAddr               = flag.String("http.listenAddr", ":8436", "http service listener addr")
+	configUpdateInterval     = flag.Duration("configUpdateInterval", time.Second*10, "interval when config will be updated")
+	percentOfUpdatingTargets = flag.Int("percentOfUpdatingTargets", 10, "percent of updating targets")
+	targetsCount             = flag.Int("targetsCount", 1000, "targetsCount defines how many copies of nodeexporter to add as scrape targets. affects metrics volume and cardinality")
+	targetName               = flag.String("targetName", "vm-benchmark-exporter.default.svc:9102", "target name with host and port name")
+	scrapInterval            = flag.Duration("scrapInterval", time.Second*5, "defines how frequently scrape targets")
+)
+
+func main() {
+	envflag.Parse()
+	logger.Init()
+	ctx := context.Background()
+	logger.Infof("starting config updater service")
+
+	if *percentOfUpdatingTargets > 100 || *percentOfUpdatingTargets < 0 {
+		logger.Panicf("percentOfUpdatingTargets can't be lower than 0 and more than 100")
+	}
+
+	c := models.InitConfigManager(models.NewConfig(
+		models.WithGlobalConfig(*scrapInterval),
+		models.WithScrapeConfig(*targetsCount, *percentOfUpdatingTargets, *targetName)))
+
+	r := runner.New(func(ctx context.Context) error {
+		return c.Update()
+	})
+
+	if err := r.Run(*configUpdateInterval); err != nil {
+		logger.Fatalf("failed to run vmagent config updater: %s", err)
+	}
+
+	go httpserver.Serve(*listenAddr, controllers.Init(ctx, *configUpdateInterval))
```

Comment:
seems like `controllers` package isn't needed. It would be simpler to move http handler fn to the main.go and delete the rest of the code.

Reply:
removed controller, and simplified part of code